packages/core/admin/admin/src/StrapiApp.js

```diff
@@ -208,7 +208,7 @@ class StrapiApp {
     if (this.customConfigurations?.locales) {
       this.configurations.locales = [
         'en',
-        ...this.customConfigurations.locales?.filter(loc => loc !== 'en'),
+        ...(this.customConfigurations.locales?.filter((loc) => loc !== 'en') || []),
```

Comment:
I think it is, because `this.customConfigurations.locales` might be falsy (hence the optional chaining). If it is, you could run into `[ ...null ]`, which is invalid. Once you do `[...(arr || [])]` you make sure it always works.

Reply:
Exactly the code was not 100% safe :)

Reply:
Isn't it `undefined` when there is `?`?

Reply:
Same for `undefined:

<img width="557" alt="Screenshot 2022-08-11 at 11 52 43" src="https://user-images.githubusercontent.com/2244375/184108407-8f01f9be-7daa-47b0-b0c9-2c2d4900754f.png">
`

The destructured value needs to implement an iterable.

Reply:
Interesting indeed ! 
![Capture d’écran 2022-08-11 à 16 02 23](https://user-images.githubusercontent.com/11708220/184151455-ddcdbd9c-9f38-4a78-87ae-a901e4d94175.png)